cstr/src/lib.rs

```diff
@@ -166,20 +167,88 @@ pub fn to_char_ptr(s: &str) -> (*mut Char, Vec<u8>) {
 }
 
 ///
-/// write_to_buffer writes the input string to the output buffer, and returns the number of bytes written
+/// write_string_to_buffer writes the input string to the output buffer, and returns the number of bytes written
 ///
 /// # Safety
 /// This writes to a raw c-pointer, which requires unsafe operations
-pub unsafe fn write_to_buffer(message: &str, len: usize, output_ptr: *mut WideChar) -> u16 {
+pub unsafe fn write_string_to_buffer(message: &str, len: usize, output_ptr: *mut WideChar) -> u16 {
     let len = std::cmp::min(message.len(), len - 1);
     let mut v = to_widechar_vec(&message[..len]);
     v.push(0);
     unsafe {
-        copy_nonoverlapping(v.as_mut_ptr(), output_ptr, len);
+        copy_nonoverlapping(v.as_ptr(), output_ptr, len);
```

Comment:
I didn't create this function, but I assume this makes more sense.

Reply:
yep!